src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java

```diff
@@ -82,6 +82,15 @@ public boolean displayDuring(State state) {
         return false;
     }
 
+    public boolean cannotRideThrough() {
+        return alert.getEffectDetails().equals("SUSPENSION");
+    }
+
+    public boolean cannotAlightOrBoard() {
+        String[] noBoarding = {"SUSPENSION", "STATION_CLOSURE", "STOP_CLOSURE", "DOCK_CLOSURE"};
+        return Arrays.asList(noBoarding).contains(alert.getEffectDetails());
```

Comment:
I like this split up, but have a proposal for a different implementation. For these `informed_entities`, there's an `activities` field:
```
# alert 119875
 {
            "activities": [
              "BOARD",
              "EXIT"
            ],
            "route": "14",
            "route_type": 3,
            "stop": "64"
          },
```
This says that the alert affects boarding and alighting, but not riding through (RIDE). Rather than special casing these effect details, you can look at the existing effects DETOUR and NO_SERVICE, and combine that with the activities to see whether it affects boarding, alighting, or riding.

Reply:
Do you know if single alert can contain informed entities with different set of activities (e.g. "BOARD" for one stop and "BOARD", "RIDE', "EXIT" for another)?

Reply:
I thought it was possible, but I don't see it in the data. The main thing that I can think of is something like a shuttle from Harvard to MGH.
- Harvard inbound: affects BOARD, RIDE
- Harvard outbound: affects RIDE, EXIT
- MIT (both): affects BOARD, RIDE, EXIT
- MGH inbound: affects RIDE, EXIT
- MGH outbound: affects BOARD, RIDE

I don't think this is actually how the shuttle is represented in the data, however.

Reply:
Refactored, please take a look.